src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -289,13 +327,104 @@ bool UnixNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     return true;
 }
 
+// Convert the return kind that was encoded by RyuJIT to the
+// enum used by the runtime.
+GCRefKind GetGcRefKind(ReturnKind returnKind)
+{
+    static_assert((GCRefKind)ReturnKind::RT_Scalar == GCRK_Scalar, "ReturnKind::RT_Scalar does not match GCRK_Scalar");
```

Comment:
Unix x64 ABI can return values in two registers, and so there are variants like `RT_Scalar_Obj`.

Do we need to handle them here? Also, it may be interesting to look into why you have not seen this problem in the tests.

Reply:
The lab testing that is currently enabled for this on Unix is fairly limited, I believe, and I have a limited capacity to stress-test the change locally so we may miss some cases. 

The static assert tests just some cases, but `GCRefKind` translation is basically one-to-one, not even sure the assert helps much here.
I am not changing GC decoding, so it may just work if it is the same data just stored/read at different location.

Reply:
Perhaps more expanded libs testing will provide failures and repro cases. 
I am not yet sure what I'd be adding here for multivalue returns.

Reply:
`Thread::InternalHijack` needs to be fixed to hande the new cases.

You can try to build a targeted repro that has a lot of methods that return `struct { int a; object b; }`. Also, check the generated code to make sure that this struct is indeed returned in `rax` and `rdx` registers (`object b` should be in rdx).

Reply:
Will take a look. Thanks!

I kind of assumed that since we can stackwalk in scenarios that already work, we may already be handling all kinds of cases, but this is a new scenario.